flask_resty/sorting.py

```diff
@@ -128,36 +129,67 @@ def get_request_field_orderings(self, view):
 class Sorting(FieldSortingBase):
     """A sorting component that allows the user to specify sort fields.
 
-    For example, to allow users to sort by `name` and/or `date`, specify the
-    following in your view::
+    For example, to allow users to sort by `title` and/or `content_length`,
+    specify the following in your view::
 
-        sorting = Sorting('name', 'date')
+        sorting = Sorting(
+            'title',
+            content_length=sql.func.length(Post.content)
+        )
 
-    One or both of `name` or `date` can be formatted in the `sort_arg` request
-    parameter to determine the sort order. For example, users can sort requests
-    by `name` ascending and `date` descending by making a ``GET`` request to::
+    One or both of `title` or `content_length` can be formatted in the
+    `sort_arg` request parameter to determine the sort order. For example,
+    users can sort requests by `name` ascending and `date` descending by
+    making a ``GET`` request to::
 
-        /api/widgets?sort=name,-date
+        /api/post?sort=title,-content_length
 
     :param str field_names: The fields available for sorting.
+        Names should match a column on your View model.
     :param str default: If provided, specifies a default sort order when the
         request does not specify an explicit sort order.
+    :param dict kwargs: Provide custom sort behavior by mapping a sort
+        argument name to a model order_by expression.
     """
 
     #: The request parameter from which the formatted sorting fields will be
     #: retrieved.
     sort_arg = "sort"
 
-    def __init__(self, *field_names, default=None):
-        self._field_names = frozenset(field_names)
+    def __init__(self, *field_names, default=None, **kwargs):
+        keys = frozenset(kwargs.keys())
+        names = frozenset(field_names)
+        duplicates = keys.intersection(names)
+
+        assert not (
+            duplicates
+        ), f"Sort field(s): {', '.join(duplicates)} cannot be passed as field names and keyword arguments"
```

Comment:
suggestion: `ValueError` should be raised for invalid arguments.

nitpick: Since this is a developer-facing error, I don't think there's a need to format it as a comma-delimited list.

```suggestion
    if duplicates:
        raise ValueError(f"Sort field(s) cannot be passed as both positional and keyword arguments: {duplicates}")
```